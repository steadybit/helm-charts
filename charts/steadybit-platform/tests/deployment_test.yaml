templates:
  - deployment.yaml
chart:
  appVersion: 0.0.0
  version: 0.0.0
tests:
  - it: manifest should support Recreate strategy
    chart:
      version: 0.0.0
    asserts:
      - matchSnapshot: { }
  - it: manifest should support RollingUpdate strategy
    chart:
      version: 0.0.0
    set:
      platform:
        rolloutStrategy: RollingUpdate
    asserts:
      - matchSnapshot: {}
  - it: manifest should support extra certificate volume
    chart:
      version: 0.0.0
    set:
      platform:
        extraVolumes:
          - name: extra-cert
            configMap:
              name: self-signed-ca
    asserts:
      - matchSnapshot: {}
  - it: manifest should support extra volumes and mounts
    chart:
      version: 0.0.0
    set:
      platform:
        extraVolumes:
          - name: test-volume
            hostPath: /test
        extraVolumeMounts:
          - name: test-volume
            mountPath: /test
    asserts:
      - matchSnapshot: {}
  - it: manifest should support init container
    chart:
      version: 0.0.0
    set:
      platform:
        database:
          migration:
            useInitContainer: true
    asserts:
      - matchSnapshot: {}
  - it: manifest should support various variables
    chart:
      version: 0.0.0
    set:
      platform:
        publicWebUrl: https://example.steadybit.com
        ingressOrigin: https://example.ingress.steadybit.com
        name: "test-stage"
    asserts:
      - matchSnapshot: {}