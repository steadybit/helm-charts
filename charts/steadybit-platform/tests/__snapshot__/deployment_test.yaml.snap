manifest should not set terminationGracePeriodSeconds:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts: null
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          serviceAccountName: RELEASE-NAME-steadybit-platform
manifest should set terminationGracePeriodSeconds:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_MAX_WAIT_TIME
                  value: 43200s
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_CHECK_INTERVAL
                  value: 11s
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts: null
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          serviceAccountName: RELEASE-NAME-steadybit-platform
          terminationGracePeriodSeconds: 43200
manifest should support Recreate strategy:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_CHECK_INTERVAL
                  value: 10s
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts: null
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          serviceAccountName: RELEASE-NAME-steadybit-platform
manifest should support RollingUpdate strategy:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        rollingUpdate:
          maxSurge: 2
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_CHECK_INTERVAL
                  value: 10s
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts: null
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          serviceAccountName: RELEASE-NAME-steadybit-platform
manifest should support extra certificate volume:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_CHECK_INTERVAL
                  value: 10s
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts: null
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          serviceAccountName: RELEASE-NAME-steadybit-platform
          volumes:
            - configMap:
                name: self-signed-ca
              name: extra-cert
manifest should support extra volumes and mounts:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_CHECK_INTERVAL
                  value: 10s
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts:
                - mountPath: /test
                  name: test-volume
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          serviceAccountName: RELEASE-NAME-steadybit-platform
          volumes:
            - hostPath: /test
              name: test-volume
manifest should support init container:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_CHECK_INTERVAL
                  value: 10s
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_TENANT_DDL_MODE_STARTUP
                  value: none
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts: null
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          initContainers:
            - env:
                - name: JAVA_TOOL_OPTIONS
                  value: -XX:+DisableAttachMechanism
                - name: SPRING_PROFILES_ACTIVE
                  value: db-migration
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              name: steadybit-platform-db-migration
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
          serviceAccountName: RELEASE-NAME-steadybit-platform
manifest should support various variables:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      minReadySeconds: 60
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-platform
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /actuator/prometheus
            prometheus.io/port: "9090"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-platform
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: steadybit-platform-0.0.0
        spec:
          containers:
            - env:
                - name: STEADYBIT_EXPERIMENT_EXECUTION_SHUTDOWN_CHECK_INTERVAL
                  value: 10s
                - name: LOGGING_FORMAT
                  value: text
                - name: STEADYBIT_TENANT_MODE
                  value: ONPREM
                - name: STEADYBIT_TENANT_AGENTKEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-platform
                - name: STEADYBIT_TENANT_KEY
                  value: onprem
                - name: STEADYBIT_TENANT_NAME
                  value: onprem
                - name: STEADYBIT_WEB_PUBLIC_URL
                  value: https://example.steadybit.com
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_PORT
                  value: "80"
                - name: STEADYBIT_WEB_PUBLIC_EXPERIMENT_URL
                  value: https://example.ingress.steadybit.com
                - name: SPRING_DATASOURCE_USERNAME
                  value: steadybit
                - name: SPRING_DATASOURCE_URL
                  value: jdbc:postgresql://RELEASE-NAME-postgresql:5432/steadybit
                - name: SPRING_DATASOURCE_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: password
                      name: RELEASE-NAME-postgresql
                - name: STEADYBIT_PLATFORM_NAME
                  value: test-stage
                - name: STEADYBIT_AUTH_PROVIDER
                  value: static
                - name: STEADYBIT_AUTH_STATIC_0_USERNAME
                  value: admin
                - name: STEADYBIT_AUTH_STATIC_0_PASSWORD
                  value: '{noop}admin'
              envFrom: []
              image: docker.steadybit.io/steadybit/platform:0.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-platform
              ports:
                - containerPort: 8080
                  name: ui-port
                  protocol: TCP
                - containerPort: 7878
                  name: websocket-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /api/health/readiness
                  port: ui-port
                initialDelaySeconds: 10
                periodSeconds: 5
                successThreshold: 1
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 4000m
                  memory: 4Gi
                requests:
                  cpu: 2000m
                  memory: 2Gi
              startupProbe:
                failureThreshold: 200
                httpGet:
                  path: /api/health/liveness
                  port: ui-port
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              volumeMounts: null
          imagePullSecrets:
            - name: RELEASE-NAME-steadybit-platform-pull-secrets
          serviceAccountName: RELEASE-NAME-steadybit-platform
